### PR TITLE
Bugfix/1190 logical model resource resolver extensions

### LINKED ISSUE #1190: FindStructureDefinitionForCoreType does not account for logical model data types
FindStructureDefinitionForCoreType does currently append the official FHIR canonical prefix ("http://hl7.org/fhir/StructureDefinition/") to all datatypes. This behaviour does not account for logical models where the type may already be represented by an absolute URL (See StructureDefinition.type: "Absolute URLs are only allowed in logical models").

### PR DESCRIPTION
**Pull request by alexzautke**
This PR fixes #1190

### CODE REVIEW

**Comment by marcovisserFurore** [comment]
Why do you alter `FindStructureDefinitionForCoreType(` for this? You could also use `FindStructureDefinition()` . So like this

`var sd = resolver.FindStructureDefinition("http://hl7.org/fhir/cda/StructureDefinition/ANY")`

**Comment by alexzautke** [comment]
This issue only occurs for complex non-FHIR data types. In this particular case ```FindStructureDefinitionForCoreType``` being called by the Snapshot generator.

**Comment by marcovisserFurore** [comment]
The problem I have is that the method `FindStructureDefinitionForCoreType` does not cover the name anymore. 

We can also alter the method `FindStructureDefinition()` and check if the parameter `uri` is indeed an uri. If not, then call the method `FindStructureDefinitionForCoreType`. This involves also a change in the class `SnapshotGenerator.getStructureForContentReference` then

File: src/Hl7.Fhir.Specification/Specification/Source/ResourceResolverExtensions.cs [hunk 1]
```diff
@@ -47,7 +47,16 @@ public static StructureDefinition FindStructureDefinition(this IResourceResolver
 
         public static StructureDefinition FindStructureDefinitionForCoreType(this IResourceResolver resolver, string typename)
         {
-            var url = ModelInfo.CanonicalUriForFhirCoreType(typename);
+            string url;
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification/Specification/Source/ResourceResolverExtensions.cs]: I would prefer the following style:
`var url = Uri.IsWellFormedUriString(typename, UriKind.Absolute) ? typename : ModelInfo.CanonicalUriForFhirCoreType(typename)`

But technically it is exact the same, so you can decide what to do with that..

Comment (alexzautke) [reply src/Hl7.Fhir.Specification/Specification/Source/ResourceResolverExtensions.cs]: Changed as requested

**Pull request merged**